scene/main/node.cpp

```diff
@@ -462,10 +462,15 @@ void Node::set_pause_mode(PauseMode p_mode) {
 	}
 
 	bool prev_inherits = data.pause_mode == PAUSE_MODE_INHERIT;
+	bool prev_can_process = is_inside_tree() && can_process();
 	data.pause_mode = p_mode;
 	if (!is_inside_tree()) {
 		return; //pointless
 	}
+	// Changing pause_mode directly can result in pausing/unpausing
+	if (prev_can_process != can_process()) {
+		notification(can_process() ? NOTIFICATION_UNPAUSED : NOTIFICATION_PAUSED);
+	}
 	if ((data.pause_mode == PAUSE_MODE_INHERIT) == prev_inherits) {
 		return; ///nothing changed
 	}
```

Comment:
This condition prevents the notification from propagating when pause mode changes between `pause` and `process` (because "inherit" state doesn't change). You can use this simple script for testing:
```GDScript
func _notification(what: int) -> void:
	if what == NOTIFICATION_PAUSED:
		prints(name, "paused")
	elif what == NOTIFICATION_UNPAUSED:
		prints(name, "processed")
```
Maybe adding a new function would indeed be better, to avoid changing the behavior too much.

Reply:
I've added a new method that propagates the notification to all descendants with inherit mode. Because some classes call `can_process` when handling pause/unpause notification, I'm calling the new method after propagating pause owner, as it can change the result.